protos/flyteidl/core/tasks.proto

```diff
@@ -118,6 +118,9 @@ message TaskTemplate {
     oneof target {
         Container container = 6;
     }
+
+    // This can be used to customize task handling at execution time for the same task type.
+    int32 task_type_version = 7;
```

Comment:
can we no call it version? or protocol_version?

Reply:
I don't think we should overload what it means... Version is confusing with task id version.. this is more about the spec...

Reply:
@kumare3 I originally had it as version but I'm inclined to agree with Haytham, this is meta info regarding the task type. Happy to change this again but not sure what you both prefer? `task_spec_version`?

Reply:
I guess we have a field called as `type` in this proto, so lets keep it `task_type_version`